box-browse-sdk/src/main/java/com/box/androidsdk/browse/fragments/BoxSearchFragment.java

```diff
@@ -1,43 +1,261 @@
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Presentation, ThumbnailManager.PRESENTATION_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Spreadsheet, ThumbnailManager.SPREADSHEET_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Video, ThumbnailManager.VIDEO_EXTENSIONS_ARRAY);
+
+        mRequest = null;
+        mRecentSearches = fetchRecentSearches();
+        mRecentSearchesAdapter = new BoxRecentSearchAdapter(getActivity(), mRecentSearches, this);
+    }
+
+    @Override
+    protected int getLayout() {
+        return R.layout.box_browsesdk_fragment_search;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+
+        if (mSearchRecentsListView != null) {
+            mSearchRecentsListView.setAdapter(mRecentSearchesAdapter);
+
+            // Show recents only if we are not using an old fragment
+            if (mSearchQuery == null) {
+                mSearchRecentsListView.setVisibility(View.VISIBLE);
+                mSearchRecentsListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                        String query = mRecentSearches.get(position);
+                        Activity activity = getActivity();
+                        if (activity instanceof BoxBrowseActivity) {
+                            ((BoxBrowseActivity)activity).setSearchQuery(query);
+                        }
+                    }
+                });
+            }
+        }
+
+        mSearchFiltersHeader = view.findViewById(R.id.filterResultsHeader);
+        mSearchFiltersHeader.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startFilterActivity();
+            }
+        });
+        setupSearchFiltersHeader();
+
+        return view;
+    }
+
+    protected void startFilterActivity() {
+        startActivityForResult(FilterSearchResults.newFilterSearchResultsIntent(getActivity(), mSearchFilters), REQUEST_FILTER_SEARCH_RESULTS);
+    }
+
+    private void setupSearchFiltersHeader() {
+        if (mSearchFiltersHeader == null || mSearchFilters == null) {
+            return;
+        }
+
+        if (mSearchFilters.anyFiltersSet()) {
+            mSearchFiltersHeader.findViewById(R.id.filterResultsHeaderUnset).setVisibility(View.GONE);
+            mSearchFiltersHeader.findViewById(R.id.filterResultsHeaderSet).setVisibility(View.VISIBLE);
+
+            TextView textView = (TextView) mSearchFiltersHeader.findViewById(R.id.filterResults);
+            ArrayList<String> filters = new ArrayList<String>();
+            for (BoxSearchFilters.ItemType type: BoxSearchFilters.ItemType.values()) {
```

Comment:
Define a toString() method for BoxSearchFilters.ItemType, so you don't need to switch/case in the fragment.